src/wp-includes/fonts/class-wp-font-family.php

```diff
@@ -0,0 +1,615 @@
+			// If the font face has the same fontStyle and fontWeight as an existing, continue.
+			foreach ( $new_font_faces as $font_to_compare ) {
+				if ( $new_font_face['fontStyle'] === $font_to_compare['fontStyle'] &&
+					$new_font_face['fontWeight'] === $font_to_compare['fontWeight'] ) {
+					$font_face_is_repeated = true;
+				}
+			}
+
+			if ( $font_face_is_repeated ) {
+				continue;
+			}
+
+			// If installing google fonts, download the font face assets.
+			if ( ! empty( $font_face['downloadFromUrl'] ) ) {
+				$new_font_face = $this->download_font_face_assets( $new_font_face );
+			}
+
+			// If installing local fonts, move the font face assets from
+			// the temp folder to the wp fonts directory.
+			if ( ! empty( $font_face['uploadedFile'] ) && ! empty( $files ) ) {
+				$new_font_face = $this->move_font_face_asset(
+					$new_font_face,
+					$files[ $new_font_face['uploadedFile'] ]
+				);
+			}
+
+			/*
+			 * If the font face assets were successfully downloaded, add the font face
+			 * to the new font. Font faces with failed downloads are not added to the
+			 * new font.
+			 */
+			if ( ! empty( $new_font_face['src'] ) ) {
+				$new_font_faces[] = $new_font_face;
+			}
+		}
+
+		if ( ! empty( $new_font_faces ) ) {
+			$this->data['fontFace'] = $new_font_faces;
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Gets the post for a font family.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return WP_Post|null The post for this font family object or
+	 *                      null if the post does not exist.
+	 */
+	public function get_font_post() {
+		$args = array(
+			'post_type'      => 'wp_font_family',
+			'post_name'      => $this->data['slug'],
+			'name'           => $this->data['slug'],
+			'posts_per_page' => 1,
+		);
+
+		$posts_query = new WP_Query( $args );
+
+		if ( $posts_query->have_posts() ) {
+			return $posts_query->posts[0];
+		}
+
+		return null;
+	}
+
+	/**
+	 * Gets the data for this object from the database and
+	 * sets it to the data property.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return WP_Post|null The post for this font family object or
+	 *                      null if the post does not exist.
+	 */
+	private function get_data_from_post() {
+		$post = $this->get_font_post();
+		if ( $post ) {
+			$this->data = json_decode( $post->post_content, true );
+			return $post;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Creates a post for a font family.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return int|WP_Error Post ID if the post was created, WP_Error otherwise.
+	 */
+	private function create_font_post() {
+		$post = array(
+			'post_title'   => $this->data['name'],
+			'post_name'    => $this->data['slug'],
+			'post_type'    => 'wp_font_family',
+			'post_content' => $this->get_data_as_json(),
+			'post_status'  => 'publish',
+		);
+
+		$post_id = wp_insert_post( $post );
+		if ( 0 === $post_id || is_wp_error( $post_id ) ) {
+			return new WP_Error(
+				'font_post_creation_failed',
+				__( 'Font post creation failed' )
```

Comment:
```suggestion
				__( 'Font post creation failed.' )
```
Missing `.` at the end of the error message.